flink-runtime-web/src/test/java/org/apache/flink/runtime/webmonitor/handlers/JarHandlerParameterTest.java

```diff
@@ -211,6 +218,37 @@ void testProvideJobId() throws Exception {
         assertThat(jobGraph.get().getJobID()).isEqualTo(jobId);
     }
 
+    @Test
+    void testProvideFlinkConfig() throws Exception {
+
+        REQB jarRequestBodyWithFlinkConfig = getJarRequestBodyWithFlinkConfig(FLINK_CONFIGURATION);
+        HandlerRequest<REQB> request =
+                createRequest(
+                        getJarRequestBodyWithFlinkConfig(FLINK_CONFIGURATION),
+                        getUnresolvedJarMessageParameters(),
+                        getUnresolvedJarMessageParameters(),
+                        jarWithManifest);
+
+        handleRequest(request);
+
+        Optional<JobGraph> jobGraph = getLastSubmittedJobGraphAndReset();
+
+        assertThat(jobGraph.isPresent()).isTrue();
+        JobGraph graph = jobGraph.get();
+
+        assertThat(getExecutionConfig(graph).getParallelism())
+                .isNotEqualTo(
+                        Integer.valueOf(
+                                FLINK_CONFIGURATION.get(CoreOptions.DEFAULT_PARALLELISM.key())));
```

Comment:
I don't understand why this assertion is passing. For the JarPlanHandler, sure, because it doesn't support the configuration, but why isn't this setting honored by the JarRunHandler?